src/Cardano/Address/Style/Shelley.hs

```diff
@@ -169,21 +218,68 @@ instance SoftDerivation Shelley where
             \either a programmer error, or, we may have reached the maximum \
             \number of addresses for a given wallet."
 
+instance PaymentAddress Shelley where
+    paymentAddress discrimination k = unsafeMkAddress $
+        invariantSize expectedLength $ BL.toStrict $ runPut $ do
+            putWord8 firstByte
+            putByteString (keyToHash k)
+      where
+          -- we use here the fact that payment address stands for what is named
+          -- as enterprise address, ie., address carrying no stake rights. For
+          -- rationale why we may need such addresses refer to delegation
+          -- specification - Section 3.2.3. What is important here is that the
+          -- address is composed of discrimination byte and 28 bytes hashed public key.
+          -- Moreover, it was decided that first 4 bits for enterprise address
+          -- will be `0110`. The next for bits for network discriminator are to
+          -- be established soon. TO_DO : change `firstbyte` as network
+          -- discriminant in CDDL spec is known.
+          firstByte = case discrimination of
+              RequiresNoMagic  -> 0x60
+              RequiresMagic (ProtocolMagic _pm) -> 0x61
```

Comment:
We need to figure out how the network discrimination exactly works in Shelley. I am not sure that this byron `RequiresNoMagic` / `RequiresMagic` applies anymore. As far as I understand from the CDDL spec, it's now just a number, so I assume it's always present.

Reply:
So,
```
data NetworkDiscriminantByron = RequiresNoMagic | RequiresMagic ProtocolMagic
data NetworkDiscriminantShelley = NetworkId Word8
data NetworkDiscriminant = NetworkDiscriminantByron | NetworkDiscriminantShelley
```
?

Reply:
attempted here https://github.com/input-output-hk/cardano-addresses/pull/20/commits/0a200b8867ef0b792340d4fbfce7731ba87450f7